components/script/dom/underlyingsourcecontainer.rs

```diff
@@ -55,6 +62,165 @@ impl UnderlyingSourceType {
     }
 }
 
+pub enum TeeCancelAlgorithm {
+    Cancel1Algorithm,
+    Cancel2Algorithm,
+}
+
+#[dom_struct]
```

Comment:
I don't think this needs to be a dom struct, rather it can just derive `JSTraceable, MallocSizeOf`

Reply:
Right, thanks for pointing it out. Maybe we can look later if that is necessary or if an `Rc` would do(and whether this makes any difference).